Framework/NexusGeometry/src/JSONGeometryParser.cpp

```diff
@@ -261,19 +261,18 @@ Json::Value getRoot(const std::string &jsonGeometry) {
 std::string extractInstrumentName(const Json::Value &instrument) {
   std::string name;
   const auto &children = instrument[CHILDREN];
-  for (const auto &child : children) {
-    if (child[NAME] == NAME)
-      name = child["values"].asString();
-  }
+  const auto it =
+      std::find_if(std::cbegin(children), std::cend(children), [](const auto child) { return child[NAME] == NAME; });
+  if (it != std::cend(children))
+    name = (*it)["values"].asString();
```

Comment:
I am not completely sure, it felt inefficient to keep iterating over the container after one child was found and then try to reassign the same value in case a match has been found later. Unfortunately, in this case, the inverse iterators are not easy to implement as this is not an STL container.

Reply:
As long as there is only one occurrence of `child[NAME] == NAME` this is fine.